src/streaming/MediaPlayer.js

```diff
@@ -1662,7 +1662,7 @@ function MediaPlayer() {
     */
     /**
      * Registers a custom capabilities filter. This enables application to filter representations to use.
-     * The provided callback function shall return a boolean based on whether or not to use the representation.
+     * The provided callback function shall return either a boolean or a promise resolving to a boolean based on whether or not to use the representation.
```

Comment:
I suggest we make returning a promise mandatory with v5. If the application logic is async then the promise can simply be resolved immediately.

This would require changing `custom-capabilities-filters.html` as well

Reply:
Related to code, this does not make any difference:
There's no need to differentiate between sync an async functions since `Promise.resolve(foo)` can take both, sync functions and Promises as `foo`.

See here, where external callbacks get consumed:
```
            customCapabilitiesFilters.forEach(customFilter => {
                promises.push(new Promise(resolve => resolve(customFilter(rep))));
            });
```
https://github.com/DolbyLaboratories/dash.js/blob/6ad9818d5717b559bd4fca2f5755f22e34a2a6cb/src/streaming/utils/CapabilitiesFilter.js#L307


Of course, we can restrict the documentation to "promises" only - your choice.

And as side note, I expect people may reuse their existing functions "as is" when migrating to v5. Therefore, backward compatibility by accepting (sync) functions is not a bad thing.

Reply:
Sounds good let's leave as is then